scripts/deploy/deploy.rb

```diff
@@ -42,13 +42,37 @@ def execute_steps(steps, step_index)
           'Continue from a specified step') do |t|
     @step_index = t.to_i
   end
+
+  opts.on('--dry-run', "Don't do a real deployment, but test what would happen if you did") do |t|
+      @is_dry_run = t
+  end
+
+  opts.on('--branch BRANCH', "Branch to deploy from") do |t|
+      @branch = t
+  end
 end.parse!
 
 steps = [
+    # Prep for making changes
     method(:validate_version_number),
+    method(:ensure_clean_repo),
+    method(:pull_latest),
+    method(:switch_to_release_branch),
+
+    # Update version number
     method(:update_read_me),
     method(:update_stripe_sdk_version),
     method(:update_gradle_properties),
+    method(:create_version_bump_pr),
 ]
 
 execute_steps(steps, @step_index)
+
+if (@is_dry_run)
+    rputs "Please verify that the dry run worked as expected, then press enter to revert all changes."
```

Comment:
_What_ should the user expect to happen here?

Maybe we should give them a list go make sure x, y, z happened, and press enter when you're done.

Reply:
Sounds good! I added that they should see a version bump PR created. I'll update this as I add more steps (in follow up PRs)